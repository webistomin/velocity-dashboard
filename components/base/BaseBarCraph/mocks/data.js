export const barGraphData = {
  labels: ['April', 'May', 'June'],
  datasets: [
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Comfort',
      lineTension: 0.4,
      backgroundColor: '#2E5BFF',
      data: [60, 65, 85],
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Premium',
      backgroundColor: '#8C54FF',
      data: [70, 80, 90],
    },
  ],
};
