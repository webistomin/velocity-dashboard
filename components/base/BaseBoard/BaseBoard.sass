+b('base-board')
  +e('wrapper')
    display: grid
    grid-gap: rem(24px)
    grid-template-columns: repeat(auto-fill, minmax(200px, 1fr))

    +media($screen-lg)
      grid-template-columns: repeat(auto-fill, minmax(20%, 1fr))

  +e('heading')
    margin-bottom: rem(10px)

  +e('item')
    display: flex
    flex-direction: column
    background-color: rgb(var(--color-white))
    padding: rem(24px)
    border-radius: 5px
    position: relative
    overflow: hidden

    &::before
      content: ''
      position: absolute
      left: 0
      top: 0
      width: rem(2px)
      height: 100%

    +m('ghost')
      background-color: rgba(var(--color-lavender), 0.2)
      font-size: 0
      border: rem(1px) dashed rgb(var(--color-primary))

      &::before
        display: none

    +m('color')
      +v('default')
        &::before
          background-color: rgb(var(--color-primary))

      +v('yellow')
        &::before
          background-color: rgb(var(--color-yellow))

      +v('purple')
        &::before
          background-color: rgb(var(--color-purple))

      +v('darkturquoise')
        &::before
          background-color: rgb(var(--color-darkturquoise))

      +v('blue')
        &::before
          background-color: rgb(var(--color-blue))

    &:not(:last-of-type)
      margin-bottom: rem(8px)

  +e('item-row')
    display: flex
    align-items: baseline
    justify-content: space-between

    &:not(:last-of-type)
      margin-bottom: rem(6px)

  +e('heading')
    display: flex
    align-items: baseline

  +e('caption')
    margin-right: rem(8px)

  +e('toggle')
    display: flex
    align-items: center
    box-sizing: border-box
    padding-top: rem(35px)
    padding-bottom: rem(35px)
    width: 100%
    position: relative
    transition: color 0.2s ease-out

    &::before,
    &::after
      content: ''
      display: inline-block
      width: 100%
      height: 1px
      background-color: rgb(var(--color-gray-4))

    &:hover
      color: rgb(var(--color-primary))

  +e('toggle-text')
    position: relative
    flex-shrink: 0
    padding: 0 rem(10px)

  +e('name')
    font-weight: 500

