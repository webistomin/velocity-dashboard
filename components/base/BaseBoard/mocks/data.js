export const boardData = [
  {
    title: 'Backlog',
    color: 'blue',
    tasks: [
      {
        title: 'Spire1',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire2',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire3',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'In progress',
    color: 'purple',
    tasks: [
      {
        title: 'Spire5',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire6',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire7',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire8',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Validate',
    color: 'darkturquoise',
    tasks: [
      {
        title: 'Spire9',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire10',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire11',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire12',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Complete',
    color: 'yellow',
    tasks: [
      {
        title: 'Spire13',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire14',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire15',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire16',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
];
