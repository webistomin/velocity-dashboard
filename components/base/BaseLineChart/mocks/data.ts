import { ChartData } from 'chart.js';

export const lineGraphData: ChartData = {
  labels: [
    '',
    'Mar 1',
    'Mar 8',
    'Mar 15',
    'Mar 22',
    'Mar 29',
    'Apr 5',
    'Apr 12',
    'Apr 19',
    'Apr 26',
    'May 2',
    'May 9',
    'May 16',
    'May 23',
    'May 30',
    'Jun 5',
    'Jun 10',
    'Jun 19',
  ],
  datasets: [
    {
      label: 'Revenue',
      borderColor: '#2E5BFF',
      pointBackgroundColor: '#FFF',
      fill: true,
      lineTension: 0,
      data: [
        {
          x: 0,
          y: 8000,
        },
        {
          x: 1,
          y: 11500,
        },
        {
          x: 2,
          y: 9800,
        },
        {
          x: 3,
          y: 15100,
        },
        {
          x: 4,
          y: 23000,
        },
        {
          x: 5,
          y: 22000,
        },
        {
          x: 6,
          y: 16000,
        },
        {
          x: 7,
          y: 13800,
        },
        {
          x: 8,
          y: 16300,
        },
        {
          x: 9,
          y: 17000,
        },
        {
          x: 10,
          y: 24500,
        },
        {
          x: 11,
          y: 19000,
        },
        {
          x: 12,
          y: 12000,
        },
        {
          x: 13,
          y: 8900,
        },
        {
          x: 14,
          y: 16000,
        },
        {
          x: 15,
          y: 27000,
        },
        {
          x: 16,
          y: 22000,
        },
        {
          x: 17,
          y: 21000,
        },
      ],
    },
    {
      label: 'Trips',
      borderColor: '#8C54FF',
      pointBackgroundColor: '#FFF',
      fill: true,
      lineTension: 0,
      data: [
        {
          x: 0,
          y: 4800,
        },
        {
          x: 1,
          y: 5000,
        },
        {
          x: 2,
          y: 7500,
        },
        {
          x: 3,
          y: 7000,
        },
        {
          x: 4,
          y: 11500,
        },
        {
          x: 5,
          y: 13000,
        },
        {
          x: 6,
          y: 7500,
        },
        {
          x: 7,
          y: 8500,
        },
        {
          x: 8,
          y: 6000,
        },
        {
          y: 13700,
        },
        {
          x: 10,
          y: 18000,
        },
        {
          x: 11,
          y: 17600,
        },
        {
          x: 12,
          y: 2500,
        },
        {
          x: 13,
          y: 6000,
        },
        {
          x: 14,
          y: 9000,
        },
        {
          x: 15,
          y: 22500,
        },
        {
          x: 16,
          y: 14000,
        },
        {
          x: 17,
          y: 19000,
        },
      ],
    },
  ],
};
