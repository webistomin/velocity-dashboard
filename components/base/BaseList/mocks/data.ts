export const listData = [
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
  {
    image: '/img/avatar.png',
    name: 'Bebop',
    text: 'Volvo Intellisafe',
    statistic: '1,232 miles',
    price: '$6,432',
  },
];
