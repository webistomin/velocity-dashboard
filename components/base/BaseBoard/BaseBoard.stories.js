import { object } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import BaseBoard from './BaseBoard';
import { boardData } from './mocks/data';

export default {
  title: 'BaseBoard',
  parameters: {
    options: {
      showPanel: true,
      selectedPanel: 'storybook/knobs/panel',
      previewTabs: {
        actions: {
          hidden: true,
        },
      },
    },
  },
};

export const Default = () => ({
  components: { BaseBoard },
  props: {
    board: {
      type: Object,
      default: object('board', boardData),
      required: true,
    },
  },
  methods: {
    onSetBoard: action('emit onSetBoard'),
  },
  template: `<base-board
              :board='this.board'
              @setBoard='this.onSetBoard'
            >
            </base-board>
  `,
});
