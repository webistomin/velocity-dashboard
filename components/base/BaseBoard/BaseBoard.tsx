import { VueComponent } from 'types/vue-components';
import { Component, Emit, Prop } from 'nuxt-property-decorator';
import { VNode } from 'vue';
import Draggable from 'vuedraggable';

import BaseBadge from 'components/base/BaseBadge';

import './BaseBoard.sass';

export interface ITask {
  title: string;
  text: string;
  date: string;
}

export interface ITaskBoard {
  title: string;
  color?: string;
  tasks: ITask[];
}

export interface IBaseBoardProps {
  board: ITaskBoard[];
  onSetBoard?: (board: ITaskBoard[]) => void;
}

@Component({
  name: 'BaseBoard',
  components: { Draggable },
})
export default class BaseBoard extends VueComponent<IBaseBoardProps> {
  @Prop({ required: true })
  private readonly board!: IBaseBoardProps['board'];

  public currentBoard = this.board;
  public collapsed = true;

  @Emit('setBoard')
  public setBoard(): IBaseBoardProps['board'] {
    return this.currentBoard;
  }

  public toggleBoard() {
    this.collapsed = !this.collapsed;
  }

  public render(): VNode {
    return (
      <div class='base-board'>
        <div class='base-board__wrapper'>
          {this.currentBoard.map((column) => {
            return (
              <div class='base-board__col'>
                <div class='base-board__heading'>
                  <strong class='base-board__caption caption'>{column.title}</strong>
                  <BaseBadge color={column.color}>{column.tasks.length}</BaseBadge>
                </div>
                <Draggable
                  data-jest='base-block__draggable'
                  class='base-board__draggable'
                  group='tasks'
                  animation={200}
                  disabled={false}
                  ghostClass='base-board__item_ghost'
                  list={column.tasks}
                  onChange={this.setBoard}>
                  {column.tasks.slice(0, this.collapsed ? 4 : column.tasks.length).map((task) => {
                    return (
                      <div class={`base-board__item base-board__item_color_${column.color || 'default'}`}>
                        <div class='base-board__item-row'>
                          <strong class='base-board__name'>{task.title}</strong>
                        </div>
                        <div class='base-board__item-row'>
                          <span class='base-board__text'>{task.text}</span>
                          <time class='base-board__date'>{task.date}</time>
                        </div>
                      </div>
                    );
                  })}
                </Draggable>
              </div>
            );
          })}
        </div>
        <button class='base-board__toggle btn caption' type='button' onClick={this.toggleBoard}>
          <span class='base-board__toggle-text'>{this.collapsed ? 'Expand' : 'Collapse'} boards</span>
        </button>
      </div>
    );
  }
}
