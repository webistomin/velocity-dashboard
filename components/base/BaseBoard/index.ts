import BaseBoard from './BaseBoard';

export default BaseBoard;
