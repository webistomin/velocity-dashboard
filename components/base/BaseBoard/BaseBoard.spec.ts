import { mount } from '@vue/test-utils';
import BaseBoard from './BaseBoard';
import { boardData } from './mocks/data';

describe('BaseBoard', () => {
  const draggableSelector = '.base-board .base-board__col .base-board__draggable';
  const factory = (propsData = {}, options = {}) => {
    return mount(BaseBoard, {
      propsData: {
        board: boardData,
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Emit setBoard event', () => {
    const wrapper = factory();
    wrapper.vm.setBoard();
    expect(wrapper.emitted().setBoard);
  });

  it('Toggle collapsed state correctly', () => {
    const wrapper = factory(
      {},
      {
        data() {
          return {
            collapsed: true,
          };
        },
      }
    );
    wrapper.vm.toggleBoard();
    expect(wrapper.vm.collapsed).toEqual(false);

    wrapper.vm.toggleBoard();
    expect(wrapper.vm.collapsed).toEqual(true);
  });

  it('Have 4 element is column if is collapsed', () => {
    const wrapper = factory(
      {},
      {
        data() {
          return {
            collapsed: true,
          };
        },
      }
    );

    const draggableEl = wrapper.find(draggableSelector);
    const draggableItems = draggableEl.findAll('.base-board__item');
    expect(draggableItems.length).toEqual(4);
  });

  it('Have more than 4 element is column if is not collapsed', () => {
    const wrapper = factory(
      {},
      {
        data() {
          return {
            collapsed: false,
          };
        },
      }
    );

    const draggableEl = wrapper.find(draggableSelector);
    const draggableItems = draggableEl.findAll('.base-board__item');
    expect(draggableItems.length).toBeGreaterThanOrEqual(4);
  });
});
