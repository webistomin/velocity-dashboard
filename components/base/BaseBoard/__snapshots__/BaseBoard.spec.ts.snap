// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseBoard Matches snapshot 1`] = `
"<div class=\\"base-board\\">
  <div class=\\"base-board__wrapper\\">
    <div class=\\"base-board__col\\">
      <div class=\\"base-board__heading\\"><strong class=\\"base-board__caption caption\\">Backlog</strong><span class=\\"base-badge base-badge_color_blue  \\">6</span></div>
      <div class=\\"base-board__draggable\\">
        <div class=\\"base-board__item base-board__item_color_blue\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire1</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_blue\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire2</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_blue\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire3</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_blue\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire4</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
      </div>
    </div>
    <div class=\\"base-board__col\\">
      <div class=\\"base-board__heading\\"><strong class=\\"base-board__caption caption\\">In progress</strong><span class=\\"base-badge base-badge_color_purple  \\">4</span></div>
      <div class=\\"base-board__draggable\\">
        <div class=\\"base-board__item base-board__item_color_purple\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire5</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_purple\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire6</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_purple\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire7</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_purple\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire8</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
      </div>
    </div>
    <div class=\\"base-board__col\\">
      <div class=\\"base-board__heading\\"><strong class=\\"base-board__caption caption\\">Validate</strong><span class=\\"base-badge base-badge_color_darkturquoise  \\">4</span></div>
      <div class=\\"base-board__draggable\\">
        <div class=\\"base-board__item base-board__item_color_darkturquoise\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire9</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_darkturquoise\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire10</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_darkturquoise\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire11</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_darkturquoise\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire12</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
      </div>
    </div>
    <div class=\\"base-board__col\\">
      <div class=\\"base-board__heading\\"><strong class=\\"base-board__caption caption\\">Complete</strong><span class=\\"base-badge base-badge_color_yellow  \\">4</span></div>
      <div class=\\"base-board__draggable\\">
        <div class=\\"base-board__item base-board__item_color_yellow\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire13</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_yellow\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire14</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_yellow\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire15</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
        <div class=\\"base-board__item base-board__item_color_yellow\\">
          <div class=\\"base-board__item-row\\"><strong class=\\"base-board__name\\">Spire16</strong></div>
          <div class=\\"base-board__item-row\\"><span class=\\"base-board__text\\">Tire replacement</span><time class=\\"base-board__date\\">June 12</time></div>
        </div>
      </div>
    </div>
  </div><button type=\\"button\\" class=\\"base-board__toggle btn caption\\"><span class=\\"base-board__toggle-text\\">Expand boards</span></button>
</div>"
`;
